### PR TITLE
All schools - enhancements and improvements

### PR DESCRIPTION
**Pull request by abe-shek**
Added - 
 - search based on programs offered for a high school, like "humanities" will list all the schools offering humanities as a program
 - search based on the address (including zip code) of the high 
 - a map link to open a new tab with google maps location for the current school 

Updated -
 - save high school page under admin application to increase security (accessible now only by superuser)

Comment - 
 - git is showing that there are some commits missing from develop (mostly all of them are automatic merge commits from develop)
 - the additions and deletions are the true representation of what the commits for this PR actually are (changes across 9 files)

### CODE REVIEW

File: high_school/templates/high_school/index.html [hunk 1]
```diff
@@ -161,19 +161,25 @@
...
+                        if (is_fav_on) {
+                            show_toast("The page will reload in 5 seconds", true);
+                            setTimeout(function () {
+                                window.location.reload();
+                            }, 5000);
+                        }
                     }
                 },
                 error: function (data) {
+                    console.log(data)
```
Comment (snehalkenjale) [inline high_school/templates/high_school/index.html]: Remove this log, or just display a text message, as data may contain any private information.

**Comment by abe-shek** [comment]
> The code looks good to me.
> 
> Some few points to go over.
> 
> 1. I want to confirm if the search filter will query only program name or content as well.
> 2. I think there's a bug concerning the filter functionality.
>    If I select a school to view its details, and then change the filter, the list gets updated but the details of the selected school are still visible although the school is filtered out.
>    The right pane should also be refreshed when filters are applied.
> 
> Let me know if you are not able to reproduce this at your end.

----
> 1. I want to confirm if the search filter will query only program name or content as well.

I confirmed this by doing a quick check by using the first school "47 The American..." - having "graduates" and "sign language" as search queries, the results will only contain the school for the latter (program name string) and will show empty results for the former (a word in program description of that school). 

> 2. I think there's a bug concerning the filter functionality.
>    If I select a school to view its details, and then change the filter, the list gets updated but the details of the selected school are still visible although the school is filtered out.
>    The right pane should also be refreshed when filters are applied.

This is a rather intended behavior of how I thought a user might interact with the overview and list views, while I was coding this functionality. The right pane serves as a brief overview of a selected school, listing its details. Now since a certain amount of user's time would be spent applying filters and browsing through schools, it struck me that keeping the overview constant during the transitions and updating it once the user settles on a school again would be more suitable. 
Please do let me know what you think about this.

File: high_school/templates/high_school/index.html (hunk 1 above)
Comment (abe-shek) [reply high_school/templates/high_school/index.html]: I keep forgetting to remove this

**Comment by snehalkenjale** [comment]
> > The code looks good to me.
> > Some few points to go over.
> > 
> > 1. I want to confirm if the search filter will query only program name or content as well.
> > 2. I think there's a bug concerning the filter functionality.
> >    If I select a school to view its details, and then change the filter, the list gets updated but the details of the selected school are still visible although the school is filtered out.
> >    The right pane should also be refreshed when filters are applied.
> > 
> > Let me know if you are not able to reproduce this at your end.
> 
> > 1. I want to confirm if the search filter will query only program name or content as well.
> 
> I confirmed this by doing a quick check by using the first school "47 The American..." - having "graduates" and "sign language" as search queries, the results will only contain the school for the latter (program name string) and will show empty results for the former (a word in program description of that school).
> 
> > 1. I think there's a bug concerning the filter functionality.
> >    If I select a school to view its details, and then change the filter, the list gets updated but the details of the selected school are still visible although the school is filtered out.
> >    The right pane should also be refreshed when filters are applied.
> 
> This is a rather intended behavior of how I thought a user might interact with the overview and list views, while I was coding this functionality. The right pane serves as a brief overview of a selected school, listing its details. Now since a certain amount of user's time would be spent applying filters and browsing through schools, it struck me that keeping the overview constant during the transitions and updating it once the user settles on a school again would be more suitable.
> Please do let me know what you think about this.

As discussed with the product owner, we can keep this functionality as it is.

**Pull request merged**